pkg/runctl/run_manager.go

```diff
@@ -112,42 +115,44 @@ func (c *runManager) prepareRunNamespace(pipelineRun k8s.PipelineRun) error {
 
 	pipelineCloneSecretName, imagePullSecretNames, err := c.copySecretsToRunNamespace(pipelineRun, runNamespace)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	if err = c.setupServiceAccount(runNamespace, pipelineCloneSecretName, imagePullSecretNames); err != nil {
-		return err
+	saSecretName, err := c.setupServiceAccount(runNamespace, pipelineCloneSecretName, imagePullSecretNames)
+	if err != nil {
+		return "", err
 	}
 
 	if err = c.setupStaticNetworkPolicies(runNamespace); err != nil {
-		return err
+		return "", err
 	}
 
-	return nil
+	return saSecretName, nil
 }
 
-func (c *runManager) setupServiceAccount(runNamespace string, pipelineCloneSecretName string, imagePullSecrets []string) error {
+func (c *runManager) setupServiceAccount(runNamespace string, pipelineCloneSecretName string, imagePullSecrets []string) (string, error) {
```

Comment:
Not sure about make this secret name a return value of this function. That is completely arbitrary. Isn't there any other way for the caller to get this.
If not it would be better to return (a copy of) the service account.

Reply:
Changed this to return the service account wrap.